File: tests/xcalablemp/global-view/gmove/Fortran/error/gmove_lc_1a1t_bc.f90
program tgmove
integer i,n=8
integer a(n),b(n)
!$xmp nodes p(2)
!$xmp template tx(n)
!$xmp distribute tx(cyclic(2)) onto p
!$xmp align a(i) with tx(i)
!$xmp align b(i) with tx(i)

!$xmp loop (i) on tx(i)
do i=1,n
  a(i)=i
end do

!$xmp loop (i) on tx(i)
do i=1,n
  b(i)=0
end do

!$xmp gmove
b(1:n)=a(1:n)

ierr=0
!$xmp loop (i) on tx(i)
do i=1,n
  ierr=ierr+abs(b(i)-a(i))
end do

print *, 'max error=',ierr

contains
end

File: tests/xcalablemp/global-view/gmove/C/gmove_lc_1a1t_bc.c
#include <stdio.h>
#include <stdlib.h>

extern int chk_int(int ierr);

int n=8;
int a[n],b[n];
#pragma xmp nodes p(2)
#pragma xmp template tx(0:n-1)
#pragma xmp distribute tx(cyclic(2)) onto p
#pragma xmp align a[i] with tx(i)
#pragma xmp align b[i] with tx(i)

int main(){

  int i,ierr;

#pragma xmp loop (i) on tx(i)
  for(i=0;i<n;i++){
    a[i]=i+1;
  }

#pragma xmp loop (i) on tx(i)
  for(i=0;i<n;i++){
    b[i]=0;
  }

#pragma xmp gmove
  b[0:n]=a[0:n];

  ierr=0;
#pragma xmp loop (i) on tx(i)
  for(i=0;i<n;i++){
    ierr=ierr+abs(b[i]-a[i]);
  }

#pragma xmp reduction (MAX:ierr)
  chk_int(ierr);
}

